dataplane/neighbor.cpp

```diff
@@ -140,6 +154,44 @@ static void netlink_neighbor_monitor(const std::function<void(const std::string&
 	close(nl_socket);
 }
 
+static void netlink_neighbor_dump(const std::function<void(const std::string&, const common::ip_address_t&, const common::mac_address_t&)>& callback)
+{
+	int nl_socket = socket(AF_NETLINK, SOCK_RAW, NETLINK_ROUTE);
+	if (nl_socket < 0)
+	{
+		YANET_LOG_ERROR("socket(): %s\n", strerror(errno));
+		return;
+	}
+
+	struct
+	{
+		nlmsghdr nl_msg;
+		ndmsg nd_msg;
+		char buf[256];
+	} request;
+
+	memset(&request, 0, sizeof(request));
+	request.nl_msg.nlmsg_len = NLMSG_LENGTH(sizeof(ndmsg));
+	request.nl_msg.nlmsg_type = RTM_GETNEIGH;
+	request.nl_msg.nlmsg_flags = NLM_F_DUMP | NLM_F_REQUEST;
+
+	static uint32_t sequence = time(nullptr);
```

Comment:
Global variables? Then we have possible race condition while doing RMW operation below.